packages/table/src/table.tsx

```diff
@@ -1191,8 +1197,10 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         return this.state.verticalGuides != null || this.state.horizontalGuides != null;
     }
 
-    private isSelectionModeEnabled(selectionMode: RegionCardinality) {
-        return this.props.selectionModes.indexOf(selectionMode) >= 0;
+    private isSelectionModeEnabled(selectionMode: RegionCardinality, selectionModes = this.props.selectionModes) {
+        const { numRows } = this.props;
+        const numColumns = this.childrenArray.length;
```

Comment:
nit: isn't there a react helper method for this? https://reactjs.org/docs/react-api.html#reactchildrencount

Reply:
Yes, there is. Didn't know which was the preferred way. Will change.